frontend/packages/topology/src/layouts/DagreLayout.ts

```diff
@@ -69,13 +76,23 @@ class DagreEdge {
   }
 }
 
+export type LayoutCallback = (nodes: Node[], edges: Edge[]) => void;
+type AdditionalDagreLayoutOptions = { onLayout: LayoutCallback };
 export default class DagreLayout implements Layout {
```

Comment:
All changes for `DagreLayout` were rushed and not really thought through. @jeff-phillips-18 @christianvogt Feel free to comment on direction / changes. I'll likely conflict with your work Jeff when it comes to your topology work.

Reply:
We can reconcile once the changes are in, you can use the `GRAPH_LAYOUT_END_EVENT` event once its there.